Objective-C/CBLDatabase.mm

```diff
@@ -144,6 +144,15 @@ - (instancetype) initWithName: (NSString*)name
         _queryQueue = dispatch_queue_create(qName.UTF8String, DISPATCH_QUEUE_SERIAL);
         
         _state = kCBLDatabaseStateOpened;
+        
+        C4Error c4error = {};
+        C4Collection* c4col = c4db_getDefaultCollection(_c4db, &c4error);
+        if (c4error.code != 0) {
+            CBLWarn(Database, @"%@ : Error getting the default collection: %d/%d",
+                    self, c4error.domain, c4error.code);
+        }
+        
+        _defaultCollection = [[CBLCollection alloc] initWithDB: self c4collection: c4col];
```

Comment:
The reason that this solves the problem because it doesn't put the returned collection into the main auto releasepool of the test which will be drained after the teardown. I don't have a good clean solution for this. If you want to use the original code, you can do it this way. I'm fine either way.

```
- (void) initDefaultCollection {
    // Note: initialize _defaultCollection under an auto releasepool to prevent the test failure
    // when checking the c4object leaks in teardown. Depending on where the database object
    // is initialized, the returned collection could be put into the main auto releasepool of
    // the test, which is drained after the teardown.
    @autoreleasepool {
        NSError* error = nil;
        _defaultCollection = [self collectionWithName: kCBLDefaultCollectionName
                                                scope: kCBLDefaultScopeName
                                                error: &error];
        if (error) {
            CBLWarn(Database, @"%@ : Error when getting the default collection: %@", self, error);
        }
    }
}
```